From 237e5d4f2fe93c72a1b09332d73cb52d871d220d Mon Sep 17 00:00:00 2001
From: Koutheir Attouchi <koutheir@gmail.com>
Date: Mon, 8 Jun 2026 11:06:57 -0400
Subject: [PATCH] Implement `bytelist2string()` manually.

`bytelist2string()` was removed from `elftools.common.utils`. Implement it in this module.
---
 dwex/dwarfone.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dwex/dwarfone.py b/dwex/dwarfone.py
index bcfc7dd..2d3224d 100644
--- a/dwex/dwarfone.py
+++ b/dwex/dwarfone.py
@@ -6,7 +6,7 @@
 from elftools.dwarf.dwarfinfo import DwarfConfig, DebugSectionDescriptor
 from elftools.dwarf.die import AttributeValue
 from elftools.dwarf.structs import DWARFStructs
-from elftools.common.utils import struct_parse, bytelist2string
+from elftools.common.utils import struct_parse
 from elftools.dwarf.enums import ENUM_DW_TAG, ENUM_DW_AT, ENUM_DW_FORM
 from elftools.construct import CString
 from elftools.dwarf.lineprogram import LineProgramEntry, LineState
@@ -274,6 +274,12 @@ def get_entries(self):
     def __getitem__(self, name):
         return self.header[name]
 
+def bytelist2string(bytelist):
+    """ Convert a list of byte values (e.g. [0x10 0x20 0x00]) to a bytes object
+        (e.g. b'\x10\x20\x00').
+    """
+    return b''.join(bytes((b,)) for b in bytelist)
+
 class DWARFExprParserV1(object):
     def __init__(self, structs):
         self.structs = structs
